go/libraries/doltcore/sqle/tables.go

```diff
@@ -611,10 +620,34 @@ func (t *WritableDoltTable) GetAutoIncrementValue(ctx *sql.Context) (interface{}
 	if !t.autoIncCol.AutoIncrement {
 		return nil, sql.ErrNoAutoIncrementCol
 	}
-	if t.ed != nil {
-		return t.ed.GetAutoIncrementValue()
+
+	return t.getAutoIncrementValue(ctx)
+}
+
+func (t *WritableDoltTable) getAutoIncrementValue(ctx *sql.Context) (interface{}, error) {
+	ed, err := t.getTableEditor(ctx)
+	if err != nil {
+		return nil, err
 	}
-	return t.DoltTable.GetAutoIncrementValue(ctx)
+
+	trackerVal, err := ed.aiTracker.Next(t.tableName)
+	if errors.Is(err, autoincr.ErrTableNotInitialized) {
```

Comment:
This has a race

Consider two sessions both hitting this line and error at the same time

The first one calls Init, Next

Then the second calls Init, Next

Now they both have the same key